duckduckgo/macos-browser
# Update BSK moving InternalUserDecider + FeatureFlagger

Task/Issue URL: https://app.asana.com/0/72649045549333/1203676877763836/f
Tech Design URL: https://app.asana.com/0/481882893211075/1203938569856702/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/270
CC:

**Description**:

This PR:
- moves the InternalUserDecider into BSK
- updates BSK to linked version (note: currently pointing to branch for review)

To understand why I ended up at the solution I did, it's worth pointing out some requirements:
- We should be able to remotely toggle features using the privacy config
- Remote toggling shouldn't necessarily be possible while the feature is in development, only when we're close to release. Up until this point, we should stick with having them enabled internally alone by authenticating as an internal user.
- The current state of a given feature (particularly during development) may well be different in iOS vs macOS and we need a way to express that.
- Some features (autofill included) may be developed (and in the future disabled) only partially, introducing the need for nested features ("subfeatures") in the Privacy Configuration.

**Steps to test this PR**:
1. Check that any features configured using the privacy-config still work as expected (particularly autofill for iOS).
2. Check that internal updates and internal app icon works

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
